src/quicklist.c

```diff
@@ -315,7 +315,11 @@ REDIS_STATIC void __quicklistCompress(const quicklist *quicklist,
         if (forward == node || reverse == node)
             in_depth = 1;
 
-        if (forward == reverse)
+        /* To make quicklist->len no matter odd or even exit early.
+         * If quicklist->len == quicklist->compress * 2, the old code will
+         * not exit here, so the middle two nodes will be compressed
+         * again. */
```

Comment:
I think there's no reason to mention old code here. Maybe something like:
```suggestion
        /* We passed into compress depth of opposite side of the quicklist 
         * so there's no need to compress anything and we can exit. */
```